src/main/java/org/jabref/gui/duplicationFinder/DuplicateResolverDialog.java

```diff
@@ -55,6 +57,12 @@ public DuplicateResolverDialog(BibEntry one, BibEntry two, DuplicateResolverType
     }
 
     private void init(BibEntry one, BibEntry two, DuplicateResolverType type) {
+        Rectangle2D primaryScreenBounds = Screen.getPrimary().getVisualBounds();
+        this.setX(primaryScreenBounds.getMinX());
+        this.setY(primaryScreenBounds.getMinY());
+        this.setWidth(primaryScreenBounds.getWidth());
+        this.setHeight(primaryScreenBounds.getHeight());
```

Comment:
This always maximizes the dialog.  IMHO, the dialog position should be changed if the new position would be not completely in the screen / completely visible. Other than that, the "old" size should be used. I am not sure how to implement that - maybe let this handled by the operating system and no additional code by JabRef.